### ISSUE TITLE
Sord doesn't handle Hash syntax like Hash{KeyType=>ValueType} correctly

### ISSUE THREAD

**Issue by connorshea** [AaronC81/sord#3]
For example, from [discordrb](https://github.com/meew0/discordrb):

```ruby
# @return [Hash<String => Change>, RoleChange, nil] lorem ipsum
attr_reader :changes
```

Sord creates this:

```ruby
# sord warn - "String => Change" does not appear to be a type
# sord warn - nil is probably not a type, but using anyway
sig { params().returns(T.any(T::Hash[SORD_ERROR_StringChange], RoleChange, nil)) }
def changes() end
```

- [Relevant discordrb code](https://github.com/meew0/discordrb/blob/329045797fb2eb76275d47403eb08016458910e6/lib/discordrb/data/audit_logs.rb#L85) 
- [Generated discordrb YARD docs for `changes`](https://meew0.github.io/discordrb/master/Discordrb/AuditLogs/Entry.html#changes-instance_method)

[YARD docs explanation of the syntax](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#Hashes):

>Hashes can be specified either via the parametrized type discussed above, in the form `Hash<KeyType, ValueType>`, or using the hash specific syntax: `Hash{KeyTypes=>ValueTypes}`. In the latter case, `KeyTypes` or `ValueTypes` can also be a list of types separated by commas.

It doesn't explicitly mention `Hash<KeyType=>ValueType>` but it seems like that's valid YARD as well.

**Comment by connorshea** [comment]
Assuming I don't get to fixing this myself, here are two simple tests for this :P
```ruby
expect(subject.yard_to_sorbet('Hash{String=>Symbol}')).to eq 'T::Hash[String, Symbol]'
expect(subject.yard_to_sorbet('Hash<String=>Symbol>')).to eq 'T::Hash[String, Symbol]'
```

**Comment by connorshea** [comment]
Oh man, I didn't realize this was being parsed with Regex 😰 

This'll be fun...

**Comment by AaronC81** [comment]
It's parsed partially by regex and partially by [a very wonky hand-written parser](https://github.com/AaronC81/sord/blob/master/lib/sord/type_converter.rb#L25) for nested generic parameters... sorry, it's not very nice!

**Comment by AaronC81** [comment]
It might be possible to adapt [this project by the author of YARD](https://github.com/lsegal/yard-types-parser) into what we want...

**Comment by connorshea** [comment]
I actually already got the regex working, I just need to figure out how to get the parser to accept `=>` as a separator 🤔

**Comment by AaronC81** [comment]
Fixed by #18!
